app/process/handler/crypto-transfer/crypto-transfer.go

```diff
@@ -1,3 +1,113 @@
 package cryptotransfer
 
-// TODO: Crypto Transfer event handler
+import (
+	"encoding/hex"
+	"fmt"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/golang/protobuf/proto"
+	"github.com/hashgraph/hedera-sdk-go"
+	hederaClient "github.com/limechain/hedera-eth-bridge-validator/app/clients/hedera"
+	"github.com/limechain/hedera-eth-bridge-validator/app/domain/repositories"
+	"github.com/limechain/hedera-eth-bridge-validator/app/services/fees"
+	"github.com/limechain/hedera-eth-bridge-validator/app/services/signer/eth"
+	"github.com/limechain/hedera-eth-bridge-validator/config"
+	protomsg "github.com/limechain/hedera-eth-bridge-validator/proto"
+	log "github.com/sirupsen/logrus"
+)
+
+// Crypto Transfer event handler
+type CryptoTransferHandler struct {
+	topicID         hedera.ConsensusTopicID
+	ethSigner       *eth.Signer
+	hederaClient    *hederaClient.HederaNodeClient
+	transactionRepo repositories.TransactionRepository
+}
+
+func (cth *CryptoTransferHandler) Handle(payload []byte) error {
+	var ctm protomsg.CryptoTransferMessage
+	err := proto.Unmarshal(payload, &ctm)
+	if err != nil {
+		return err
+	}
+
+	exists, err := cth.transactionRepo.Exists(ctm.TransactionId)
+	if err != nil {
+		log.Error(fmt.Sprintf("Error while trying to get transaction [%s] from database", ctm.TransactionId))
+		return err
+	}
+
+	if exists {
+		return nil
+	}
+
+	err = cth.transactionRepo.Create(&ctm)
+	if err != nil {
```

Comment:
I think that we should log in both cases:
- when we create the new transaction in the DB
- when we tried to create it but there was an error ( I know that there is a logger when we return this `err` but we won't have the context `when` the error occurred